src/main/java/io/stargate/sgv2/jsonapi/api/model/command/builders/CollectionSortClauseBuilder.java

```diff
@@ -77,25 +84,44 @@ public SortClause buildAndValidate(ObjectNode sortNode) {
       return SortClause.mutable(SortExpression.collecetionVectorizeSort(vectorizeData));
     }
 
-    // Otherwise, use shared default processing
-    return super.buildAndValidate(sortNode);
+    // Otherwise, only "regular" sort expressions are left
+    // So let's validate the paths for the sort expressions
+    validateSortExpressionPaths(sortNode);
+
+    Iterator<Map.Entry<String, JsonNode>> fieldIter = sortNode.fields();
+    List<SortExpression> sortExpressions = new ArrayList<>(sortNode.size());
+
+    while (fieldIter.hasNext()) {
+      Map.Entry<String, JsonNode> inner = fieldIter.next();
+      sortExpressions.add(buildSortExpression(inner.getKey(), inner.getValue()));
+    }
+    return new SortClause(sortExpressions);
   }
 
-  @Override
-  protected void validateSortClausePath(String path) {
-    super.validateSortClausePath(path);
+  private void validateSortExpressionPaths(ObjectNode sortNode) {
+    Iterator<String> it = sortNode.fieldNames();
+    while (it.hasNext()) {
+      validateSortExpressionPath(it.next());
+    }
+  }
+
+  private void validateSortExpressionPath(String path) {
     if (!NamingRules.FIELD.apply(path)) {
-      // Fail on empty (blank) and "$"-starting names (conflict with operators),
-      // except allow "well-known" fields
-      switch (path) {
-        case DocumentConstants.Fields.LEXICAL_CONTENT_FIELD,
-            DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD,
-            DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD -> {}
-        default ->
-            throw ErrorCodeV1.INVALID_SORT_CLAUSE_PATH.toApiException(
-                "path ('%s') cannot start with '$' (except for pseudo-fields '$lexical', '$vector' and '$vectorize')",
-                path);
+      // This is only called after handling "special" sort expressions so simple validation
+      if (path.isEmpty()) {
```

Comment:
checking if this should be isBlank()

Reply:
Apparently not: I realized that we had decided to allow blank but NOT empy name for Collections, based on comments. Not sure what was the reason for allowing all-blank -- possibly since we allow white space in names in general.

Can revisit, but for now wanted to make handling consistent with current rules.